Build failure: gfxreconstruct

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

build gfxreconstruct

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/316557584

### Relevant log output

fails in hydra due to incompatible cmake version, once you fix the cmake problem, you get:

```console

gfxreconstruct> /build/source/framework/util/json_util.cpp: In function 'void gfxrecon::util::FieldToJson(nlohmann::json_v3_11_0::ordered_json&, std::wstring_view, const JsonOptions&)':
gfxreconstruct> /build/source/framework/util/json_util.cpp:186:10: error: 'template<class _Codecvt, class _Elem, class _Wide_alloc, class _Byte_alloc> class std::__cxx11::wstring_convert' is deprecated [-Werror=deprecated-declarations]
gfxreconstruct>   186 |     std::wstring_convert<std::codecvt_utf8<wchar_t>> utf8_conv;
gfxreconstruct>       |          ^~~~~~~~~~~~~~~
gfxreconstruct> In file included from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/locale:47,
gfxreconstruct>                  from /nix/store/qarrb8yfby1yyypm32vabzgxgq3w41ma-gcc-15.2.0/include/c++/15.2.0/iomanip:53,
gfxreconstruct>                  from /build/source/framework/util/to_string.h:32,
gfxreconstruct>                  from /build/source/framework/util/json_util.h:32,
gfxreconstruct>                  from /build/source/framework/util/json_util.cpp:24:
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2-zen2, NixOS, 26.05 (Yarara), 26.05.19700101.dirty`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`


### Notify maintainers


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)

@Flakebi 

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/gf/gfxreconstruct/package.nix
+++ b/pkgs/by-name/gf/gfxreconstruct/package.nix
@@ -28,6 +28,14 @@ stdenv.mkDerivation rec {
     fetchSubmodules = true;
   };
 
+  cmakeFlags = [
+    # The CMakeLists.txt is actually 3.10 compatible, but it specifies 3.5 as `CMAKE_VERSION_MINIMUM`
+    "-DCMAKE_POLICY_VERSION_MINIMUM=3.10"
+  ];
+
+  # Workaround for "error: ... class std::__cxx11::wstring_convert' is deprecated [-Werror=deprecated-declarations]"
+  env.NIX_CFLAGS_COMPILE = "-Wno-error=deprecated-declarations";
+
   buildInputs = [
     libX11
     libxcb

